network/transport/udp.go

```diff
@@ -0,0 +1,105 @@
+/*
+ *    Copyright 2018 Insolar
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package transport
+
+import (
+	"bytes"
+	"net"
+
+	"github.com/insolar/insolar/log"
+	"github.com/insolar/insolar/network/transport/packet"
+	"github.com/insolar/insolar/network/transport/relay"
+	"github.com/pkg/errors"
+)
+
+type udpTransport struct {
+	baseTransport
+	serverConn net.PacketConn
+}
+
+func newUDPTransport(conn net.PacketConn, proxy relay.Proxy, publicAddress string) (*udpTransport, error) {
+	transport := &udpTransport{
+		baseTransport: newBaseTransport(proxy, publicAddress),
+		serverConn:    conn}
+	transport.sendFunc = transport.send
+
+	return transport, nil
+}
+
+func (udpT *udpTransport) send(recvAddress string, data []byte) error {
+
+	log.Debug("Sending PURE_UDP request")
+
+	// TODO: may be try to send second time if error
+	// TODO: skip resolving every time by caching result
+	udpAddr, err := net.ResolveUDPAddr("udp", recvAddress)
+	if err != nil {
+		return errors.Wrap(err, "udpTransport.send")
+	}
+
+	udpConn, err := net.DialUDP("udp", nil, udpAddr)
+	if err != nil {
+		return errors.Wrap(err, "udpTransport.send")
+	}
+	defer udpConn.Close()
+
+	log.Debug("udpTransport.send: len = ", len(data))
+	_, err = udpConn.Write(data)
+	return errors.Wrap(err, "Failed to write data")
+}
+
+// Start starts networking.
+func (udpT *udpTransport) Start() error {
+	log.Info("Start UDP transport")
+	for {
+		buf := make([]byte, 1400)
```

Comment:
move MTU to constants (or even configurable options) and return error when we try to send packet with size more than MTU

Reply:
fixed